### ISSUE TITLE
https

### ISSUE THREAD

**Issue by alexmihajlov** [pjhampton/kibana-prometheus-exporter#272]
Please add setting for https if kibana only works with it.

**Comment by alexmihajlov** [comment]
I am using kibana 7.10.1 in the docker along with the installed kibanaPrometheusExporter plugin.

When I make a query 
` curl https://user:password@127.0.0.1:5601/_prometheus/metrics`

I get an error:
```
{"type":"log","@timestamp":"2022-05-06T14:32:10Z","tags":["error","http"],"pid":8,"message":"{ Error: socket hang up\n    at createHangUpError (_http_client.js:332:15)\n    at Socket.socketOnEnd (_http_client.js:435:23)\n    at Socket.emit (events.js:203:15)\n    at endReadableNT (_stream_readable.js:1145:12)\n    at process._tickCallback (internal/process/next_tick.js:63:19)\n  code: 'ECONNRESET',\n  config:\n   { url: 'http://127.0.0.1:5601/api/status',\n     method: 'get',\n     headers:\n      { Accept: 'application/json, text/plain, */*',\n        Authorization: 'Basic emFiYml4OnpieDEyMzQ1',\n        'User-Agent': 'axios/0.21.1' },\n     transformRequest: [ [Function: transformRequest] ],\n     transformResponse: [ [Function: transformResponse] ],\n     timeout: 0,\n     adapter: [Function: httpAdapter],\n     xsrfCookieName: 'XSRF-TOKEN',\n     xsrfHeaderName: 'X-XSRF-TOKEN',\n     maxContentLength: -1,\n     maxBodyLength: -1,\n     httpsAgent:\n      Agent {\n        _events: [Object],\n        _eventsCount: 1,\n        _maxListeners: undefined,\n        defaultPort: 443,\n        protocol: 'https:',\n        options: [Object],\n        requests: {},\n        sockets: {},\n        freeSockets: {},\n        keepAliveMsecs: 1000,\n        keepAlive: false,\n        maxSockets: Infinity,\n        maxFreeSockets: 256,\n        maxCachedSessions: 100,\n        _sessionCache: [Object] },\n     validateStatus: [Function: validateStatus],\n     data: undefined },\n  request:\n   Writable {\n     _writableState:\n      WritableState {\n        objectMode: false,\n        highWaterMark: 16384,\n        finalCalled: false,\n        needDrain: false,\n        ending: false,\n        ended: false,\n        finished: false,\n        destroyed: false,\n        decodeStrings: true,\n        defaultEncoding: 'utf8',\n        length: 0,\n        writing: false,\n        corked: 0,\n        sync: true,\n        bufferProcessing: false,\n        onwrite: [Function: bound onwrite],\n        writecb: null,\n        writelen: 0,\n        bufferedRequest: null,\n        lastBufferedRequest: null,\n        pendingcb: 0,\n        prefinished: false,\n        errorEmitted: false,\n        emitClose: true,\n        autoDestroy: false,\n        bufferedRequestCount: 0,\n        corkedRequestsFree: [Object] },\n     writable: true,\n     _events:\n      [Object: null prototype] {\n        response: [Function: handleResponse],\n        error: [Function: handleRequestError] },\n     _eventsCount: 2,\n     _maxListeners: undefined,\n     _options:\n      { maxRedirects: 21,\n        maxBodyLength: 10485760,\n        protocol: 'http:',\n        path: '/api/status',\n        method: 'GET',\n        headers: [Object],\n        agent: undefined,\n        agents: [Object],\n        auth: undefined,\n        hostname: '127.0.0.1',\n        port: '5601',\n        nativeProtocols: [Object],\n        pathname: '/api/status' },\n     _ended: true,\n     _ending: true,\n     _redirectCount: 0,\n     _redirects: [],\n     _requestBodyLength: 0,\n     _requestBodyBuffers: [],\n     _onNativeResponse: [Function],\n     _currentRequest:\n      ClientRequest {\n        _events: [Object],\n        _eventsCount: 7,\n        _maxListeners: undefined,\n        output: [],\n        outputEncodings: [],\n        outputCallbacks: [],\n        outputSize: 0,\n        writable: true,\n        _last: true,\n        chunkedEncoding: false,\n        shouldKeepAlive: false,\n        useChunkedEncodingByDefault: false,\n        sendDate: false,\n        _removedConnection: false,\n        _removedContLen: false,\n        _removedTE: false,\n        _contentLength: 0,\n        _hasBody: true,\n        _trailer: '',\n        finished: true,\n        _headerSent: true,\n        socket: [Socket],\n        connection: [Socket],\n        _header:\n         'GET /api/status HTTP/1.1\\r\\nAccept: application/json, text/plain, */*\\r\\nAuthorization: Basic emFiYml4OnpieDEyMzQ1\\r\\nUser-Agent: axios/0.21.1\\r\\nHost: 127.0.0.1:5601\\r\\nConnection: close\\r\\n\\r\\n',\n        _onPendingData: [Function: noopPendingOutput],\n        agent: [Agent],\n        socketPath: undefined,\n        timeout: undefined,\n        method: 'GET',\n        insecureHTTPParser: undefined,\n        path: '/api/status',\n        _ended: false,\n        res: null,\n        aborted: undefined,\n        timeoutCb: null,\n        upgradeOrConnect: false,\n        parser: null,\n        maxHeadersCount: null,\n        _redirectable: [Circular],\n        [Symbol(isCorked)]: false,\n        [Symbol(outHeadersKey)]: [Object] },\n     _currentUrl: 'http://127.0.0.1:5601/api/status' },\n  response: undefined,\n  isAxiosError: true,\n  toJSON: [Function: toJSON] }"}
{"type":"error","@timestamp":"2022-05-06T14:32:10Z","tags":[],"pid":8,"level":"error","error":{"message":"Internal Server Error","name":"Error","stack":"Error: Internal Server Error\n    at HapiResponseAdapter.toInternalError (/usr/share/kibana/src/core/server/http/router/response_adapter.js:69:19)\n    at Router.handle (/usr/share/kibana/src/core/server/http/router/router.js:177:34)\n    at process._tickCallback (internal/process/next_tick.js:68:7)"},"url":{"protocol":null,"slashes":null,"auth":null,"host":null,"port":null,"hostname":null,"hash":null,"search":null,"query":{},"pathname":"/_prometheus/metrics","path":"/_prometheus/metrics","href":"/_prometheus/metrics"},"message":"Internal Server Error"}
```
The plugin seems to make requests to http://127.0.0.1:5601/api/status , which is not available, instead of going to https://127.0.0.1:5601/api/status. 
But from what I understand, from the log information `defaultPort: 443,\n        protocol: 'https:'` about the schema is transmitted.

**Comment by pjhampton** [comment]
Thanks for the bug report, and the patience @alexmihajlov. There is a couple of things here.

1. You are correct. There was a TLS bug fixed here: https://github.com/pjhampton/kibana-prometheus-exporter/commit/15bccc31c9bc416b52d4d81cab3f6582a4a59bfe. This was caused by a move from Axios to node-fetch, to align with how Kibana is evolving when making internal / external requests. I will release this fix in `7.17.4+` and `8.1.3+`

2. You are using an old version of the Elastic stack. Please upgrade to benefit from security and perf fixes, and new features.

----

For future me:

```sh
$ mkcert localhost 127.0.0.1 ::1
```

```yml
server.ssl.enabled: true
server.ssl.certificate: /Users/ph/workspace/kibana/example.com+5.pem
server.ssl.key: /Users/ph/workspace/kibana/example.com+5-key.pem
```

```sh
$ https https://localhost:5601/_prometheus/metrics
```
